caddyhttp/proxy/proxy_test.go

```diff
@@ -913,6 +913,68 @@ func TestHostSimpleProxyNoHeaderForward(t *testing.T) {
 	}
 }
 
+func TestReverseProxyTransparentHeaders(t *testing.T) {
+	testCases := []struct{
+		name string
+		remoteAddr string
+		forwardedForHeader string
+		expected []string
+	} {
+		{"No header", "192.168.0.1:80", "", []string{"192.168.0.1"}},
+		{"Existing", "192.168.0.1:80", "1.1.1.1, 2.2.2.2", []string{"1.1.1.1, 2.2.2.2, 192.168.0.1"}},
```

Comment:
2 Test cases. One for no existing **X-Forwarded-For** header and one with existing